bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/api/RestUtils.java

```diff
@@ -0,0 +1,193 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.lgthinq.internal.api;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.math.BigInteger;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import javax.ws.rs.core.UriBuilder;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentProvider;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.FormContentProvider;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.util.Fields;
+import org.openhab.core.i18n.CommunicationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RestUtils} rest utilities
+ *
+ * @author Nemer Daud - Initial contribution
+ */
+@NonNullByDefault
+public class RestUtils {
+
+    private static final Logger logger = LoggerFactory.getLogger(RestUtils.class);
+    private static final String HMAC_SHA1_ALGORITHM = "HmacSHA1";
+
+    public static String getPreLoginEncPwd(String pwdToEnc) {
+        MessageDigest digest;
+        try {
+            digest = MessageDigest.getInstance("SHA-512");
+        } catch (NoSuchAlgorithmException e) {
+            logger.error("Definitively, it is unexpected.", e);
+            throw new IllegalStateException("Unexpected error. SHA-512 algorithm must exists in JDK distribution", e);
+        }
+        digest.reset();
+        digest.update(pwdToEnc.getBytes(StandardCharsets.UTF_8));
+
+        return String.format("%0128x", new BigInteger(1, digest.digest()));
+    }
+
+    public static byte[] getOauth2Sig(String messageSign, String secret) {
+        byte[] secretBytes = secret.getBytes(StandardCharsets.UTF_8);
+        SecretKeySpec signingKey = new SecretKeySpec(secretBytes, HMAC_SHA1_ALGORITHM);
+
+        try {
+            Mac mac = Mac.getInstance(HMAC_SHA1_ALGORITHM);
+            mac.init(signingKey);
+            return Base64.getEncoder().encode(mac.doFinal(messageSign.getBytes(StandardCharsets.UTF_8)));
+        } catch (NoSuchAlgorithmException e) {
+            logger.error("Unexpected error. SHA1 algorithm must exists in JDK distribution.", e);
+            throw new IllegalStateException("Unexpected error. SHA1 algorithm must exists in JDK distribution", e);
+        } catch (InvalidKeyException e) {
+            logger.error("Unexpected error.", e);
```

Comment:
`IllegalStateException` is also thrown.
```suggestion
            logger.debug("Unexpected error.", e);
```